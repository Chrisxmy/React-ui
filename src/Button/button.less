@import '../style/var.less';

.x-button {
  display: inline-block;
  font-size: 14px;
  font-weight: 400;
  height: 32px;
  padding: 0 15px;
  white-space: nowrap;
  text-align: center;
  background-image: none;
  border-radius: 4px;
  color: rgba(0, 0, 0, 0.65);
  outline: none;
  user-select: none;
  cursor: pointer;
  border: 1px solid @grey;
  box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2);
  transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
  background: @white;
  &:focus,
  &:active {
    outline: none;
  }
  &:hover {
    color: @blue-hover;
    border-color: @blue-hover;
  }
  &-primary {
    color: @white;
    background-color: @blue;
    border: 1px solid transparent;
    &:hover {
      color: @white;
      background-color: @blue-hover;
      border-color: @blue-hover;
    }
  }
  &-dashed {
    border: 1px dashed @grey;
    background: transparent;
  }
  &-danger {
    color: @white;
    background-color: @red;
    border: 1px solid transparent;
    &:hover {
      color: @white;
      background-color: @red-hover;
      border-color: @red-hover;
    }
  }
  &-link {
    color: @blue;
    border: none;
    box-shadow: none;
    background: @white;
    &:hover {
      color: @blue-hover;
    }
  }
}
