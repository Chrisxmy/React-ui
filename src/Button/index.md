---
nav:
  title: Components
  path: /components
group:
  title: Basic
  path: /basic
---

## Button æŒ‰é’®

Demo:

```tsx
import React from 'react';
import { Button } from 'dumi';

export default () => {
  return (
    <>
      <Button type="primary">primary</Button>
      <Button>default</Button>
      <Button type="dashed">dashed</Button>
      <Button type="danger">danger</Button>
      <Button type="link">link</Button>
    </>
  );
};
```
