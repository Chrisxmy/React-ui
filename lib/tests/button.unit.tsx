import * as React from 'react'
import * as renderer from 'react-test-renderer';
import Button from '../button'

describe('button test', () => {

    it('button 存在', () => {
        const json = renderer.create(<Button/>).toJSON()
        expect(json).toMatchSnapshot()
    })

})

