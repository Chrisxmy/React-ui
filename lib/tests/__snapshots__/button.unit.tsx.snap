// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test button 存在 1`] = `
<div>
  按钮
</div>
`;
